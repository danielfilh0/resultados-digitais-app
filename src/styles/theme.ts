const theme = {
  grid: {
    container: '123rem'
  },
  font: {
    family: {
      primary: "'Nunito Sans', sans-serif",
      secondary: "'Darker Grotesque', sans-serif",
      tertiary: "'Red Hot Display', sans-serif"
    },
    regular: 400,
    bold: 700,
    extrabold: 800,
    sizes: {
      xs: '1.2rem',
      sm: '1.4rem',
      md: '1.6rem',
      lg: '2rem',
      xlg: '2.4rem',
      '2xlg': '4rem'
    }
  },
  colors: {
    primary: {
      500: '#6C5AE0',
      700: '#4839A3',
      950: '#100C27'
    },
    secondary: {
      400: '#F4CA6D',
      500: '#F2BF4E'
    },
    gray: {
      300: '#CFD3D8',
      500: '#97A1AC',
      600: '#7E8A98'
    },
    white: '#FFFFFF',
    black: '#000000',
    light: '#F1F3F5',
    dark: '#212429'
  },
  spacings: {
    xs: '1.6rem',
    sm: '2.4rem',
    md: '3.2rem',
    lg: '4.8rem'
  }
}

export default theme
