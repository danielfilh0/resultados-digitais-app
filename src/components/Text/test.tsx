import { screen } from '@testing-library/react'

import { Text } from '.'
import theme from '@/styles/theme'
import { renderWithTheme } from '@/utils/tests/helpers'

describe('<Text />', () => {
  it('should render the white text as default ', () => {
    const { container } = renderWithTheme(<Text>Random text</Text>)

    expect(screen.getByText(/random text/i)).toHaveStyle({
      color: theme.colors.white
    })
    expect(screen.getByText(/random text/i)).toHaveStyle({
      'font-size': theme.font.sizes.xs
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the text with different sizes', () => {
    renderWithTheme(<Text size="medium">First text</Text>)
    renderWithTheme(<Text size="large">Second text</Text>)

    expect(screen.getByText(/first text/i)).toHaveStyle({
      'font-size': theme.font.sizes.sm
    })
    expect(screen.getByText(/second text/i)).toHaveStyleRule(
      'font-size',
      theme.font.sizes.lg,
      {
        media: '(min-width:  768px)'
      }
    )
  })
})
