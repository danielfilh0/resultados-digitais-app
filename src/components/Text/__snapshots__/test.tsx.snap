// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render the white text as default  1`] = `
.c0 {
  color: #FFFFFF;
  line-height: 1.7;
  font-weight: 400;
  font-size: 1.2rem;
}

<p
  class="c0"
>
  Random text
</p>
`;
